diff --git a/Source/WebCore/bindings/js/JSHTMLMediaElementCustom.cpp b/Source/WebCore/bindings/js/JSHTMLMediaElementCustom.cpp
index 61c33bf76..bf0879c2a 100644
--- a/Source/WebCore/bindings/js/JSHTMLMediaElementCustom.cpp
+++ b/Source/WebCore/bindings/js/JSHTMLMediaElementCustom.cpp
@@ -24,6 +24,8 @@
  */
 
 #include "config.h"
+
+#if ENABLE(VIDEO)
 #include "JSHTMLMediaElement.h"
 
 #include "TrackOpaqueRoot.h"
@@ -40,3 +42,5 @@ void JSHTMLMediaElement::visitAdditionalChildrenInGCThread(Visitor& visitor)
 DEFINE_VISIT_ADDITIONAL_CHILDREN_IN_GC_THREAD(JSHTMLMediaElement);
 
 } // namespace WebCore
+
+#endif // ENABLE(VIDEO)
